harvard-lts/drs-import-management-service
# ePADD application support

**ePADD application support**
* * *

**GitHub Issue**: ([link](https://app.zenhub.com/workspaces/hdc-62163f60fb08c60011fa4795/issues/harvard-lts/hdc/209))

Other Relevant Links (Mailing list discussion, related pull requests, etc.)
* [JWT setup for new DIMS depositing applications](https://docs.google.com/document/d/1x7nVLwMZipAh5c-thoYm4MUCJRAA8WLdjhdUv6X00YM)
* [API & Queue Message Definitions](https://docs.google.com/document/d/1Q3KqCRlmcyrQfQxfCYWLUAKSs9DTOj9yp3m4ovrRA7w/edit#heading=h.hreo56n7vc4a)

# What does this Pull Request do?
Adds support for ePADD depositing application, including the necessary changes in the authorization layer to support the ingest requests from this application. In addition, the changes introduced facilitate the extension to new depositing applications.

On the other hand, the following minor improvements have been added:

* Application admin_metadata sending to process queue (dims-data-ready), instead of transfer queue.
* Improved the local test environment. Two separate ActiveMQ instances have been added for functional and integration tests, in order to avoid collisions when running functional and integration tests simultaneously.

# How should this be tested?

* Make sure test .env variables are updated based on .env.example file
* Rebuild test environment: docker-compose -f docker-compose-test.yml build --no-cache
* Run the local test environment: `docker-compose -f docker-compose-test.yml up`
* Execute tests: `docker exec -it test_runner pytest test`

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests No
- integration tests No
- functional tests Yes

# Interested parties
@jessjass @ives1227 @adaybujeda 

Keyword: resolved
